### PR TITLE
allow processing all non-command messages

### PR DESCRIPTION
**Pull request by AlexDunmow**

### CODE REVIEW

File: messagerouter/response.go [hunk 1]
```diff
@@ -8,6 +8,7 @@ import (
 
 type ResponseWriter interface {
 	Write(string) error
+	WriteChannel(string, string) error
```
Comment (nii236) [inline messagerouter/response.go:11]: Your interface has changed so this will break existing tests

File: messagerouter/response.go [hunk 2]
```diff
@@ -54,6 +61,13 @@ func (w *SlackResponseWriter) WriteError(errText string) error {
 type CLIResponseWriter struct {
 }
 
+// Write writes to CLI, prints channel it would go to if via Slack
```
Comment (nii236) [inline messagerouter/response.go:64]: `// WriteChannel...`

File: messagerouter/commands.go [hunk 3]
```diff
@@ -75,14 +89,19 @@ func hasPrefix(s string) bool {
 // Handle registers the handler for the given command
 // Must match a command prefix, `!` for admin commands, `%` for general commands & `#` for MUD commands
 func (mr *MessageRouter) Handle(cmdString string, fn MessageHandler) error {
-	if !hasPrefix(cmdString) {
+	if !hasPrefix(cmdString) && cmdString != "*" {
```
Comment (nii236) [inline messagerouter/commands.go:93]: Can this command prefix "*" be added to the docstring above

**Review by nii236 — APPROVED** [review]
Good to merge after some minor fixes

File: messagerouter/commands.go (hunk 3 above)
Comment (AlexDunmow) [reply messagerouter/commands.go:93]: It's not really a command prefix, it's to tell the Handle function that it's a function to parse all messages. Perhaps needs its own handle function for clarity?

File: messagerouter/response.go (hunk 1 above)
Comment (AlexDunmow) [reply messagerouter/response.go:11]: Good catch

File: messagerouter/commands.go (hunk 3 above)
Comment (nii236) [reply messagerouter/commands.go:93]: Adding an extra line of comment to explain this is probably enough for now

Comment (AlexDunmow) [reply messagerouter/commands.go:93]: :+1:

**Pull request merged**